api/api/controllers/search_controller.py

```diff
@@ -310,88 +267,91 @@ def _resolve_index(
-        if serializer_field in search_params.data:
-            s = _apply_filter(s, search_params, serializer_field, es_field, "exclude")
-
-    # Exclude mature content and disabled sources
-    s = _exclude_sensitive_by_param(s, search_params)
-    s = _exclude_filtered(s)
+    query_filters = {
+        "filter": [
+            ("extension", None),
+            ("category", None),
+            ("source", None),
+            ("license", None),
+            ("license_type", "license"),
+            # Audio-specific filters
+            ("length", None),
+            # Image-specific filters
+            ("aspect_ratio", None),
+            ("size", None),
+        ],
+        "must_not": [
+            ("excluded_source", "source"),
+        ],
+    }
+    for behaviour, filters in query_filters.items():
+        for serializer_field, es_field in filters:
+            if not (arguments := search_params.data.get(serializer_field)):
+                continue
+            arguments = arguments.split(",")
+            parameter = es_field or serializer_field
+            queries[behaviour].append(Q("terms", **{parameter: arguments}))
+    return queries
+
+
+def create_ranking_queries(
+    search_params: media_serializers.MediaSearchRequestSerializer,
+) -> list[Q]:
+    queries = [Q("rank_feature", field="standardized_popularity", boost=DEFAULT_BOOST)]
+    if search_params.data["unstable__authority"]:
+        boost = int(search_params.data["unstable__authority_boost"] * DEFAULT_BOOST)
+        authority_query = Q("rank_feature", field="authority_boost", boost=boost)
+        queries.append(authority_query)
+    return queries
+
+
+def create_search_query(
+    search_params: media_serializers.MediaSearchRequestSerializer,
+) -> Q:
+    # Apply filters from the url query search parameters.
+    url_queries = create_search_filter_queries(search_params)
+    search_queries = {
+        "filter": url_queries["filter"],
+        "must_not": url_queries["must_not"],
+        "must": [],
+        "should": [],
+    }
+
+    # Exclude mature content
+    if not search_params.validated_data["include_sensitive_results"]:
+        search_queries["must_not"].append(Q("term", mature=True))
+    # Exclude dynamically disabled sources (see Redis cache)
+    if excluded_providers := get_dynamically_excluded_providers():
+        search_queries["must_not"].append(Q("terms", provider=excluded_providers))
 
     # Search either by generic multimatch or by "advanced search" with
     # individual field-level queries specified.
-    search_fields = ["tags.name", "title", "description"]
     if "q" in search_params.data:
         query = _quote_escape(search_params.data["q"])
         base_query_kwargs = {
             "query": query,
-            "fields": search_fields,
+            "fields": DEFAULT_SEARCH_FIELDS,
             "default_operator": "AND",
         }
 
         if '"' in query:
             base_query_kwargs["quote_field_suffix"] = ".exact"
```

Comment:
Just noticing this, but we don't have any `.exact` subfields. If we switch this to `.raw`, though, I believe it would start to work, because title, description and tags.name do have `.raw` subfields that are not analysed (and therefore not stemmed, so should theoretically be able to service exact match queries).

Separate issue though!

Reply:
That's interesting! I didn't look into what `.exact` does, just copied the existing code :) I'll open a new issue.

Reply:
Found the PR that added the `.exact` filter: [Make quoted queries behave as described in the API documentation (return exact matches only)](https://github.com/WordPress/openverse-api/pull/1012)

I'll open an issue for this.

Reply:
Issue: [Correctly set up mixing of exact search with stemming(https://github.com/WordPress/openverse/issues/3269)